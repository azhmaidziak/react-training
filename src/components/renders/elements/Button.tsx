import * as React from "react";

const Button = ({
  label,
  onClick
}: {
  label: string;
  onClick: (e: React.MouseEvent) => void;
}): React.ReactElement => <button onClick={onClick}>{label}</button>;
export default Button;
