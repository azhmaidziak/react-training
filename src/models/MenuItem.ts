export default interface MenuItem {
    readonly keyId: string;
    readonly text: string;
}