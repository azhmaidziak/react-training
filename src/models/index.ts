export enum SearchBy {
    TITLE = 0,
    GENRES = 1
}

export enum SortBy {
    REALIZE_DATE,
    RATING
}

export type SearchPaneModel = {
    query: string;
    searchBy: SearchBy;
};

export type Movie = {
    id: number;
    title: string;
    tagline: string;
    voteAverage: number;
    voteCount: number;
    releaseDate: Date;
    posterPath: string;
    overview: string;
    budget: number;
    revenue: number;
    runtime: number;
    genres: Set<string>;
};

export type Page = {
    title: string;
    top: SearchQuery | Movie;
    movies: Movie[];
};

export type Film = {
    id: number;
    title: string;
    tagline: string;
    vote_average: number;
    vote_count: number;
    release_date: string;
    poster_path: string;
    overview: string;
    budget: number;
    revenue: number;
    runtime: number;
    genres: string[];
};

export type SearchQuery = {
    query: string;
};

export enum SortOrder {
    DESC = 'desc',
    ASC = 'asc'
}

export type  SearchParameters = {
    sortBy: SortBy
    sortOrder: SortOrder
    query: string
    searchBy: SearchBy
    filter: Array<string>
    offset: number
    limit: number
}
