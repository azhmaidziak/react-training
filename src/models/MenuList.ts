import MenuItem from './MenuItem'

export default interface MenuList {
    readonly keyId: string;
    readonly items: MenuItem[];
}