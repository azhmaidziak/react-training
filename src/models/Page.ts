import MenuList from "./MenuList";

export type Body = {
    keyId: string
    bodyText: string
}
export default interface Page {
    menu: MenuList,
    body: Body
}