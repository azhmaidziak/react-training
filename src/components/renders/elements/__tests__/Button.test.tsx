import * as React from "react";
import Button from "../Button";
import { shallow } from "enzyme";
import * as renderer from "react-test-renderer";
import { ButtonType } from "../../../../models";

describe("Button Element Test", () => {
  const label = "Button";
  const type = ButtonType.button;
  const onClick = jest.fn(() => {});

  it("Test render correctly", () => {
    const testRenderer = renderer.create(
      <Button type={type} label={label} onClick={onClick} />
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it("Test click event", () => {
    const button = shallow(
      <Button type={type} label={label} onClick={onClick} />
    );
    button.find("button").simulate("click");
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
