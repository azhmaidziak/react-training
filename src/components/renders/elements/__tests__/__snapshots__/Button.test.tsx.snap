// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Element Test Test render correctly 1`] = `
<button
  onClick={[MockFunction]}
  type="button"
>
  Button
</button>
`;
